components/list/list-item-drag-mixin.js

```diff
@@ -98,3 +99,84 @@ export const ListItemDragMixin = superclass => class extends superclass {
 		` : nothing;
 	}
 };
+
+export class NewPositionEventDetails {
+	/**
+	 * @param { Object } object An simple object with the position event properties
+	 * @param { String } object.targetKey The item key of the list-item that is moving
+	 * @param { String } object.destinationKey The item key of the list-item in the position we are moving to
+	 * @param { String } object.temporaryMovement Information on whether the item is entering or exiting temporary movement
+	 */
+	constructor(object) {
+		this.targetKey = object.targetKey;
+		this.destinationKey = object.destinationKey;
+		this.temporaryMovement = object.temporaryMovement;
+	}
+
+	/**
+	 * Announces a move to screen readers on an array
+	 * @param { Array<Node> } list The array to announce the move on.
+	 * @param { function(any, Number): String } announceCallback Callback function that returns the announcement text
+	 *        Takes the item in the array and the index as arguments. Should return the text to announce.
+	 */
+	announceMove(list, announceCallback) {
+		const index = this.fetchPosition(list, this.targetKey);
+		if (index === null) throw new Error('Item to announce move not found in given array.');
+		const item = list[index];
+		const message = announceCallback(item, index);
+		if (message) announce(message);
+	}
+
+	/**
+	 * Fetches an index position within a list given an item key
+	 * @param { Array<Node> } list Array of nodes
+	 * @param { String } key Item key to fetch position of
+	 */
+	fetchPosition(list, key) {
+		const index = list.findIndex(x => x.key === key);
+		return index === -1 ? null : index;
+	}
+
+	/**
+	 * Reorders an array in place with the current event information
+	 * The item will be moved to the position of the destinationKey. Array elements shift
+	 * forward one to make room.
+	 * @param { Array<Node> } list The array to reorder
+	 * @param { function(any, Number): String } [announceCallback] Callback function that returns the announcement text
+	 *        Takes the item in the array and the index as arguments. Should return the text to announce.
+	 *        This optional callback will be passed to announceMove if given.
+	 */
+	reorder(list, announceCallback = null) {
```

Comment:
It be cool if this had the `key callback` as well. You could make the second parameter an object => `reorder(list, {announceCallBack, keyfn})` Define similar to https://lit-html.polymer-project.org/guide/template-reference#repeat